### ISSUE TITLE
On close, actually check for error payload

### ISSUE THREAD

**Issue by joekarl** [joekarl/go-libapns#17]
There's no check to see if the error payload is actually set or not here https://github.com/joekarl/go-libapns/blob/master/connection.go#L316-L328. 
Solely looking at the error code of the apple error is probably not a great idea. Need to couple with looking at the `appleError.MessageID` as well and start the `payloadIdCounter` at 1 instead of 0.

**Comment by joekarl** [comment]
@Boerworz this ought to fix what you were seeing.
If no objections I'll get this merged in.

**Comment by Boerworz** [comment]
Seems to work! I no longer receive an ErrorPayload or any UnsentPayloads when the client closes the connection.
However, on a somewhat related note, I find it a bit odd that an error is sent on the CloseChannel ([here](https://github.com/joekarl/go-libapns/blob/master/connection.go#L244-248) and then [here](https://github.com/joekarl/go-libapns/blob/master/connection.go#L336-341)) despite there not being an error response from the APNs server. In the example app, this causes the error handler to get called.

I would expect there not to be an error when the connection was closed by the client. It's easy to work around in the code that uses go-libapns (just check ErrorCode == 10 && MessageID == 0), but perhaps it would be better if no error was received from the library?
Also, if I've interpreted the [Provider Communication with Apple Push Notification Service](https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/CommunicatingWIthAPS.html) document correctly, receiving an ErrorCode of 10 one would expect to find the last successfully sent notification identifier in MessageID. This is currently not the case when the client closes the connection.

So, basically what I want is a way for the go-libapns library to tell me if the client closed the connection or if the APNs server did, ideally by not sending an error in the first scenario (or by using a different MessageID).
If I'm thinking about this in the wrong way, please let me know :)

**Comment by joekarl** [comment]
Yeah I hear you, I updated that PR to not return an AppleError on normal disconnect. 
Now will only report if an error is actually read from the socket or if the socket is closed unexpectedly.
